neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/IterationResult.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2013-2018 Atlanmod, Inria, LS2N, and IMT Nantes.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v2.0 which accompanies
+ * this distribution, and is available at https://www.eclipse.org/legal/epl-2.0/
+ */
+
+package fr.inria.atlanmod.neoemf.core.internal;
+
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.FeatureMap;
+
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import static fr.inria.atlanmod.commons.Preconditions.checkNotNull;
+
+/**
+ * A result prepared by iterating a {@link ListIterator} in a defined {@link IterationHelper#direction() direction}.
+ *
+ * @param <E> the type of the value
+ *
+ * @see ContentsListIterator
+ */
+@ParametersAreNonnullByDefault
+class IterationResult<E> {
+
+    /**
+     * The instance of an undefined {@code IterationResult}.
+     */
+    @Nonnull
+    private static final IterationResult<?> UNDEFINED = new EmptyIterationResult<>();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this reference to "EmptyIterationResult". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2390)